posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -0,0 +1,276 @@
+
+A new Handle API is added to make programatic lookup of Beans and Instances easier.
+
+In Jakarta Annotations 2.1, the `@priority` annotation can now be applied to any program element. This annotation is used by CDI 4.0 to allow Stereotypes to include a Priority annotation.
+
+The following previously deprecated CDI APIs are removed:  
+
+* `@New qualifier`, which is replaced by `@dependent` beans in CDI 1.1 and later.
+
+* `Bean#isNullable()`, which has not been used by the implementation since CDI 1.1.
+
+* `BeanManager#fireEvent()`. Use `BeanManager.getEvent()` instead in CDI 2.0 and later. 
+
+* `BeanManager#createInjectionTarget(AnnotatedType)`, which is replaced by `BeanManager#getInjectionTargetFactory(AnnotatedType)` in CDI 1.1 and later.
+
+* `BeforeBeanDiscovery#addAnnotatedType(AnnotatedType)`, which is replaced by `BeforeBeanDiscovery#addAnnotatedType(AnnotatedType, String)` in CDI 1.1 and later.
+
+* The Liberty configuration element has been superseded by a new element that applies to all versions of CDI (1.2 and newer).
+
+[source, xml]
+----
+<cdi enableImplicitBeanArchives="false" emptyBeansXmlCDI3Compatibility="true"/>
+----
+
+The `enableImplicitBeanArchives` attribute is the same as it was in previous versions.
+If set to `true`, which is the default, then archives that do not contain a `beans.xml` file are treated as Implicit Bean Archives and scanned for classes that have Bean Defining Annotations.
+If set to `false`, then archives that do not contain a `beans.xml` file are not scanned for annotated classes.
+
+The `emptyBeansXmlCDI3Compatibility` attribute applies only to CDI 4.0.
+If set to `true`, then an archive that contains an empty `beans.xml` file is treated as an Explicit Bean Archive, as it was in CDI 3.0 and earlier.
+
+If set to `false`, which is the default, then an archive that contains an empty `beans.xml` file is treated as an Implicit Bean Archive. This is equivalent to setting `bean-discovery-mode="annotated"`.
+
+[source, xml]
+----
+<featureManager>
+  <feature>cdi-4.0</feature>
+</featureManager>
+
+<cdi enableImplicitBeanArchives="false" emptyBeansXmlCDI3Compatibility="true"/>
+----
+
+[#expression]
+== Jakarta EE 10 Expression Language 5.0
+
+Jakarta EE 10 Expression Language 5.0 includes a number of new features, deprecations, and specification clarifications.  One of the new features is the addition of the link:https://jakarta.ee/specifications/expression-language/5.0/apidocs/jakarta.el/jakarta/el/methodreference[MethodReference] which "Provides information about the method to which a method expression resolves." This new class allows developers to access any annotations present on a particular method using the `getAnnotations()` method.
+
+To enable Expression Language 5.0, add the `expressionLanguage-5.0` feature to the list of features in your server.xml file:
+[source,xml]
+----
+   <featureManager>
+        <feature>expressionLanguage-5.0</feature>
+    </featureManager>
```

Comment:
```suggestion
<featureManager>
  <feature>expressionLanguage-5.0</feature>
</featureManager>
```

Reply:
There is an extra space for `</featureManager>` for this one and mpOpenAPI-3.1.  I was also trying to make them all consistent since there is inconsistent spacing in the examples.